### ISSUE TITLE
PingCAP pull requests are not linked to commits properly

### ISSUE THREAD

**Issue by joncodo** [merico-dev/lake#1135]
## Describe the bug
In the DB, I expect pull requests to be linked to commits. I am testing by sample since I do not know how to get the totals on pull_reques_commits join tables.

This is what I am seeing:

```
-- github url https://github.com/pingcap/tidb/pull/30945
-- find github id with GET https://api.github.com/repos/pingcap/tidb/pulls/30945
-- FAILED - SELECT * FROM lake.pull_request_commits where pull_request_id = 808245708;


-- github url https://github.com/pingcap/tidb/pull/19945
-- GET https://api.github.com/repos/pingcap/tidb/pulls/19945
-- FAILED - SELECT * FROM lake.pull_request_commits where pull_request_id = 484663162;
```

## To Reproduce
1. Go to https://github.com/pingcap/tidb/pull/19945
2. See the number of commits as 2 from the github UI
3. do a get request to get the github id `GET https://api.github.com/repos/pingcap/tidb/pulls/19945`
4. See the github id as 484663162 from the response
5. select from the join table to see two commits: SELECT * FROM lake.pull_request_commits where pull_request_id = 484663162;

## Actual
No commits are found

@klesh am I testing this in the wrong way? I see lots of records in all the related tables.

**Comment by joncodo** [comment]
<img width="438" alt="Screen Shot 2022-01-27 at 11 24 28 AM" src="https://user-images.githubusercontent.com/3011407/151389153-7946226b-6ce2-45fb-b687-9e4fcbaf004b.png">
<img width="835" alt="Screen Shot 2022-01-27 at 11 25 04 AM" src="https://user-images.githubusercontent.com/3011407/151389255-6b4db952-2322-42f2-8ef7-40b1102f2bf4.png">
<img width="1114" alt="Screen Shot 2022-01-27 at 11 25 24 AM" src="https://user-images.githubusercontent.com/3011407/151389315-71fac4ec-5e73-46cc-90dc-dc05e50ae1f3.png">

**Comment by joncodo** [comment]
You believe that this testing process is not right?

```Go to Revert "*: using standard error to replace terror (#19425)" pingcap/tidb#19945
See the number of commits as 2 from the github UI
do a get request to get the github id GET https://api.github.com/repos/pingcap/tidb/pulls/19945
See the github id as 484663162 from the response
select from the join table to see two commits: SELECT * FROM lake.pull_request_commits where pull_request_id = 484663162;```

I think it is right. We should do a call or something then. Can you provide an example of how you would test this @warren830 ?

**Comment by joncodo** [comment]
So maybe the right question to ask here is "is this the solution we want?"

I think this seems a bit confusing and maybe it works in our current code. I assumed that we were meant to use the pull request_id since that is what it is called in the table:

<img width="689" alt="Screen Shot 2022-01-27 at 3 27 48 PM" src="https://user-images.githubusercontent.com/3011407/151429786-8cc3bad0-1259-4ce1-875e-875dbff32f64.png">

When I follow @warren830 's advice, I do find the commits that I want. I was just not expecting to have to use issue_id instead of pull_request_id.

Can you fill me in on the history of this decision and why we made it @klesh ?

**Comment by klesh** [comment]
@joncodo Oh, man, I'm expecting you can fill me in, github plugin are basically finished by @kevin-kline and you.

**Comment by klesh** [comment]
looks like issues/prs are collected via `CollectIssues` task which is not ideal.
will be addressed via #1148
